### ISSUE TITLE
option:deselected and option:deselecting are not getting emitted

### ISSUE THREAD

**Issue by closingin** [sagalbot/vue-select#1430]
**Describe the bug**

Hello there! I've installed the latest version of vue-select (3.11.2, but also tried 3.11.1), and it seems like the two deselection events do not get emitted.

**To Reproduce**
```html
<vue-select
    :searchable="false"
    :options="options"
    :value="value"
    :reduce="option => option.value"
    @input="$emit('input', $event)"
    @option:selected="$emit('select')"
    @option:deselected="$emit('select')" />
```

**Expected behavior**
`option:deselected` should get fired. As you can see in my screenshot, the two `selecting` and `selected` events do get fired, but not deselection ones (corresponding the last `input` event you see). I've also tried with `option:deselecting`, and it also doesn't work.

**Screenshots**
![image](https://user-images.githubusercontent.com/2735603/118238588-fc8c1680-b498-11eb-993e-f34d3e7cf653.png)

**Comment by allanbenelli** [comment]
I have encountered the same problem when using a single selection. However using a multiple selection emits the two deselection events! 
![image](https://user-images.githubusercontent.com/17219792/118620025-6b34e100-b7c5-11eb-8470-4aa8c9d89c3f.png)

**Comment by closingin** [comment]
@allanbenelli Yeah I made the same change as a workaround, but I felt like it was worth opening the issue, thanks for confirming the bug though!

**Comment by closingin** [comment]
After a small investigation, the problems comes from the fact that clicking on the "clear selection" button doesn't call the same method (`clearSelection()`) as when you deselect an option in a multiple select (`deselect()`). The `option:deselect(ing|ed)` events get only emitted in the `deselect()` method, and they get passed the deselected option as parameter.

I would say that technically it's not a bug, as "clearing selection" is not the same as "deselecting something". But if we have `option:selected` then I think that we must also have `option:deselected`.

This would mean adding two `$emit` in the [`clearSelection`](https://github.com/sagalbot/vue-select/blob/master/src/components/Select.vue#L691) method, maybe with no parameter?

**Comment by ferrykranenburgcw** [comment]
Same issue here, at least `option:deselected` should be emitted when `clearSelection` was called or implement a new $emit like `option:cleared`. Now I have to implement this logic inside a watcher on a variable that works with `v-model`...

**Comment by depressedpsychiatrist** [comment]
> ```
> @input="$emit('input', $event)"
> ```

How could you explain? I am not able to handle callback of @input on single selects.

**Comment by LeeDongGeon1996** [comment]
Yeah, I got the same problem.. Implementing `option:cleared` is a great idea.

**Comment by clwilliams8** [comment]
I'm running into a new issue. How can I fix this without overwriting the base code of the package? I tried doing:
@option:deselecting="this.$emit('optionDeselecting', $event)"
@deselect="this.$emit('optionDeselecting', $event)"

but neither are firing off the event to the parent when clicking the "X" button or Deselect button

**Comment by iwasherefirst2** [comment]
Having the same issue. I am using a select via ajax, meaning when editing a form I only have one option in my list, and then deselect does not work.

**Comment by KeionneDerousselle** [comment]
> 4.0.0-beta.5 - Deselect button not running option:deselecting event #1747

I'm also experiencing this issue. I cannot listen to `@input` events

**Comment by rizkhal** [comment]
Any idea about this issue?

Only `@option:selected` event works for me

**Comment by Monty9120** [comment]
Using `@update:modelValue` worked for me. That seems to fire both when selecting and deselecting items.

**Comment by heartforit** [comment]
I had to drill trough the back of the head, but it works (very ugly workaround i guess):

````javascript
const abc = ref()
$(abc.value.$refs.clearButton).click((e) => {
  selectedOption(null)
})
````

```vue
<v-select ref="abc"
@search="findData"
@update:modelValue="selected"
v-model="selected"
@option:selected="selectedOption"
label="name"
:filterable="false"
:options="options"
></v-select>
```

**Comment by MatejLukacUP** [comment]
> Using `@update:modelValue` worked for me. That seems to fire both when selecting and deselecting items.

WORKS, THANKS

**Comment by buraksturk** [comment]
:deselectFromDropdown="true" property worked for me

**Comment by akashdubey2211** [comment]
Update **vue-select** to the latest version and set **:deselectFromDropdown="true"** to enable deselection from the dropdown.
